tests/benchmarks/memory_funcs.py

```diff
@@ -0,0 +1,172 @@
+"""Benchmark memory usage of various functions in DESC."""
+
+import gc
+import os
+import sys
+import warnings
+
+import numpy as np
+import pytest
+
+sys.path.insert(0, os.path.abspath("."))
+sys.path.append(os.path.abspath("../../"))
+
+import desc
+
+if sys.argv[2] in ["GPU", "gpu"]:
+    # Set the environment variable to use the GPU
+    os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+    from desc import set_device
+
+    set_device("gpu")
+
+import desc.examples
+from desc.backend import jax
+from desc.grid import LinearGrid
+from desc.magnetic_fields import ToroidalMagneticField
+from desc.objectives import (
+    BoundaryError,
+    EffectiveRipple,
+    FixCurrent,
+    FixPressure,
+    FixPsi,
+    ForceBalance,
+    ObjectiveFunction,
+    QuasisymmetryTwoTerm,
+    get_equilibrium_objective,
+    get_fixed_boundary_constraints,
+    maybe_add_self_consistency,
+)
+from desc.optimize import LinearConstraintProjection, ProximalProjection
+
+
+@pytest.mark.skip()
+@pytest.mark.unit
```

Comment:
probably better to just add a new mark for memory tests rather than marking them as unit + skip